### PR TITLE
[TIMOB-23757] Android: Fix TextField keyboard overlap

### PR DESCRIPTION
**Pull request by garymathews**
- Fix Android issue [182191](https://code.google.com/p/android/issues/detail?id=182191)

###### TEST CASE
```Javascript
var win = Ti.UI.createWindow({
        backgroundColor : 'white'
    }),
    a = Ti.UI.createTextField({
        bottom : 10,
        width: '90%',
        textAlign : Titanium.UI.TEXT_ALIGNMENT_RIGHT,
        color: 'black'
    });

win.add(a);
win.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-23757)

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java [hunk 1]
```diff
@@ -433,6 +436,18 @@ protected KrollDict getFocusEventObject(boolean hasFocus)
 	@Override
 	public boolean onEditorAction(TextView v, int actionId, KeyEvent keyEvent)
 	{
+		// TIMOB-23757: https://code.google.com/p/android/issues/detail?id=182191
+		if (Build.VERSION.SDK_INT < 24 && (getGravity() & Gravity.LEFT) != Gravity.LEFT) {
```
Comment (fmerzadyan) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java]: change from `getGravity()` to `v.getGravity()`?

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIEditText.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package ti.modules.titanium.ui.widget;
+
+import android.content.Context;
+import android.os.Build;
+import android.util.AttributeSet;
```
Comment (fmerzadyan) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIEditText.java]: add `import android.view.Gravity`?

Comment (fmerzadyan) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIEditText.java]: change spaces to tabs?

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java [hunk 3]
```diff
@@ -433,6 +436,18 @@ protected KrollDict getFocusEventObject(boolean hasFocus)
...
+			if (getNativeView() != null) {
+				ViewGroup view = (ViewGroup) getNativeView().getParent();
+				view.setFocusableInTouchMode(true);
+				view.requestFocus();
+			}
+			Context context = TiApplication.getInstance().getApplicationContext();
+			InputMethodManager inputManager = (InputMethodManager) context.getSystemService(Context.INPUT_METHOD_SERVICE);
+			inputManager.hideSoftInputFromWindow(tv.getWindowToken(), 0);
```
Comment (fmerzadyan) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java]: change `0` to `InputMethodManager.RESULT_UNCHANGED_SHOWN`?

**Review by frankieandone — DISMISSED** [review]
Slight compile errors to fix and some other suggestions but on the whole great job! 👍

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java (hunk 3 above)
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java]: [hideSoftInputFromWindow](https%3A%2F%2Fdeveloper.android.com%2Freference%2Fandroid%2Fview%2Finputmethod%2FInputMethodManager.html%23hideSoftInputFromWindow(android.os.IBinder%2C%20int)) `0` is used for the flags. I'm not using the `ResultReceiver`.

**Comment by ssjsamir** [comment]
@garymathews After testing this I am still seeing the problem on a Nexus 6 Device running Android 6.0.1.

**Steps taken:** 
+ Copied the test case from the description above
+ Tested the test case with the build from this PR
+ Pressed the text box and entered the text which was visible 
+ Pressed the back button the device  
+ Pressed the text box again and this time the text box was not visible

**Environment**
Appcelerator Command-Line Interface, version 6.1.0
Google Nexus 6P (Android 7.1.1)
Operating System Name: Mac OS X El Capitan
Operating System Version: 10.11.6
Node.js Version: 4.6.0
npm: 4.2.8
Xcode: 8.2
Appcelerator Studio: 4.8.1.201612050850

**Comment by garymathews** [comment]
@ssjsamir Updated PR

**Comment by ssjsamir** [comment]
FR - Passed,
@garymathews After testing again on the Nexus 6 (6.0.1),Nexus 5x (6.0.1) and Nexus 6P (7.1.1) I am no longer seeing an overlap between the text field and the keyboard. 

**Test steps**
+ Copied the test case from the description above
+ Tested the test case with the build from this PR
+ Pressed the text box and entered the text which was visible
+ Pressed the back button on the device
+ Pressed the text box again and this time the text box was not overlapping with the keyboard. 

**Environment**
Appcelerator Command-Line Interface, version 6.1.0
Google Nexus 6 (6.0.1)
Google Nexus 5x (6.0.1)
Google Nexus 6P (Android 7.1.1)
Operating System Name: Mac OS X El Capitan
Operating System Version: 10.11.6
Node.js Version: 4.6.0
npm: 4.2.8
Xcode: 8.2
Appcelerator Studio: 4.8.1.201612050850

**Pull request merged**